src/database/core/RepoManager.ts

```diff
@@ -121,12 +130,21 @@ export class RepoManager {
    * @return {!Repo} The Repo object for the specified server / repoName.
    */
   createRepo(repoInfo: RepoInfo, app: FirebaseApp): Repo {
-    let repo = safeGet(this.repos_, app.name);
+    let appRepos = safeGet(this.repos_, app.name);
+
+    if (!appRepos) {
+      appRepos = {};
+      this.repos_[app.name] = appRepos;
+    }
+
+    let repo = safeGet(appRepos, repoInfo.toURLString());
     if (repo) {
-      fatal('FIREBASE INTERNAL ERROR: Database initialized multiple times.');
+      fatal(
+        'Database initialized multiple times. Please make sure the format of the database URL matches with each database() call.'
```

Comment:
Alternatively we could just return the existing instance.  Would that cause other problems though?  I guess app would end up with multiple references to it and might call INTERNAL.delete() multiple times, etc.?

I guess I don't really feel strongly. It's probably not too harmful to make them pass the exact same URL every time.

Reply:
Yeah, I don't know that I'm terribly heartbroken making users pass the extra string around, especially considering that they can still just use the project config instead.